### ISSUE TITLE
TypeScript build throws type errors when type-checking the plugin

### ISSUE THREAD

**Issue by JstnMcBrd** [eslint-stylistic/eslint-stylistic#276]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

When writing an eslint-config in TypeScript that uses `@stylistic/eslint-plugin`, TypeScript throws lots of type errors.

This can be worked around by using `"skipLibCheck": true`.

This isn't a problem for most people who use `@stylistic/eslint-plugin`, because if the package is simply included in `eslint.config.js`, it is not type-checked. But in my case, I am trying to write a publishable eslint-config and I am writing it in TypeScript, so the plugin is type-checked.

Here are the errors:

```sh
node_modules/@stylistic/eslint-plugin-plus/dts/index.d.ts:7:46 - error TS2694: Namespace '"/home/projects/pariwrqko.github/node_modules/@types/eslint/index".ESLint' has no exported member 'RuleModule'.

7   [K in keyof UnprefixedRuleOptions]: ESLint.RuleModule
                                               ~~~~~~~~~~

node_modules/@stylistic/eslint-plugin/dist/dts/rule-options.d.ts:6:11 - error TS2320: Interface 'UnprefixedRuleOptions' cannot simultaneously extend types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions'.
  Named property ''comma-dangle'' of types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions' are not identical.

6 interface UnprefixedRuleOptions extends UnprefixedRuleOptions$1, UnprefixedRuleOptions$2, UnprefixedRuleOptions$3, UnprefixedRuleOptions$4 {}
            ~~~~~~~~~~~~~~~~~~~~~

node_modules/@stylistic/eslint-plugin/dist/dts/rule-options.d.ts:6:11 - error TS2320: Interface 'UnprefixedRuleOptions' cannot simultaneously extend types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions'.
  Named property ''keyword-spacing'' of types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions' are not identical.

6 interface UnprefixedRuleOptions extends UnprefixedRuleOptions$1, UnprefixedRuleOptions$2, UnprefixedRuleOptions$3, UnprefixedRuleOptions$4 {}
            ~~~~~~~~~~~~~~~~~~~~~

node_modules/@stylistic/eslint-plugin/dist/dts/rule-options.d.ts:6:11 - error TS2320: Interface 'UnprefixedRuleOptions' cannot simultaneously extend types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions'.
  Named property ''lines-around-comment'' of types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions' are not identical.

6 interface UnprefixedRuleOptions extends UnprefixedRuleOptions$1, UnprefixedRuleOptions$2, UnprefixedRuleOptions$3, UnprefixedRuleOptions$4 {}
            ~~~~~~~~~~~~~~~~~~~~~

node_modules/@stylistic/eslint-plugin/dist/dts/rule-options.d.ts:6:11 - error TS2320: Interface 'UnprefixedRuleOptions' cannot simultaneously extend types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions'.
  Named property ''lines-between-class-members'' of types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions' are not identical.

6 interface UnprefixedRuleOptions extends UnprefixedRuleOptions$1, UnprefixedRuleOptions$2, UnprefixedRuleOptions$3, UnprefixedRuleOptions$4 {}
            ~~~~~~~~~~~~~~~~~~~~~

node_modules/@stylistic/eslint-plugin/dist/dts/rule-options.d.ts:6:11 - error TS2320: Interface 'UnprefixedRuleOptions' cannot simultaneously extend types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions'.
  Named property ''padding-line-between-statements'' of types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions' are not identical.

6 interface UnprefixedRuleOptions extends UnprefixedRuleOptions$1, UnprefixedRuleOptions$2, UnprefixedRuleOptions$3, UnprefixedRuleOptions$4 {}
            ~~~~~~~~~~~~~~~~~~~~~


Found 6 errors in 2 files.

Errors  Files
     1  node_modules/@stylistic/eslint-plugin-plus/dts/index.d.ts:7
     5  node_modules/@stylistic/eslint-plugin/dist/dts/rule-options.d.ts:6
```

Please see the linked reproduction. It probably shows the problem better than I can explain it.

### Reproduction

https://stackblitz.com/edit/github-wqgfgt

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by bradzacher** [comment]
it's also worth noting that the plugin does not have any form of a dependency on `@types/eslint` even though its types explicitly depend on it.

So if someone doesn't install `@types/eslint` then the signature of all of the configs goes to `any`.

**Comment by JstnMcBrd** [comment]
@antfu this issue still exists with the latest update. I've updated the reproduction to test it. Could you reopen the issue?

**Comment by JstnMcBrd** [comment]
Once the changes from #283 and #284 are published in the next release, I'll test them and see if the issue is resolved.

**Comment by JstnMcBrd** [comment]
Can confirm, this issue is resolved in [1.6.3](https://github.com/eslint-stylistic/eslint-stylistic/releases/tag/v1.6.3).
